modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DBBasedExpiryMessageDeletionTask.java

```diff
@@ -0,0 +1,163 @@
+                         */
+                        if (messageStoresUnavailable != null) {
+
+                            log.info("Message store has become unavailable therefore waiting until store becomes available");
+                            //act as a barrier
+                            messageStoresUnavailable.get();
+                            log.info("Message store became available. Resuming ack handler");
+                            messageStoresUnavailable = null; // we are passing the blockade
+                            // (therefore clear the it).
+                        }
+
+                        if (null != expiredMessages && !expiredMessages.isEmpty()) {
+
+                            //in order to stop the slot delivery worker to this queue in the deletion range
+                            SlotManagerClusterMode.setCurrentDeletionQueue(queueName);
+                            SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+                            SlotManagerStandalone.setCurrentDeletionQueue(queueName);
+                            SlotManagerStandalone.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+
+                            //delete message metadata, content from the meta data table, content table and expiry table
+                            MessagingEngine.getInstance().deleteMessages(expiredMessages);
+
+                            //since the deletion task is finished that queue no need to hold the slot delivery worker for that queue
+                            SlotManagerClusterMode.setCurrentDeletionQueue("");
+                            SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(0L);
+                            SlotManagerStandalone.setCurrentDeletionQueue("");
+                            SlotManagerStandalone.setCurrentDeletionRangelowerboundID(0L);
+
+
+                            if (log.isDebugEnabled()) {
+                                log.debug("Expired message count for queue : " + queueName + "is" + expiredMessages.size());
+                            }
+                        }
+                    }
+                }
+            }
+            //sleep the message expiration worker for specified amount of time
+            // sleepForWaitInterval(workerWaitInterval);
+        } catch (AndesException e) {
+            log.error("Error running Message Expiration Checker " + e.getMessage(), e);
+        } catch (InterruptedException e) {
+            log.error("Thread interrupted while waiting for message stores to come online", e);
+        } catch (ExecutionException e) {
+            log.error("Error occurred while waiting for message stores to come online", e);
+        }
+    }
+
+    @Override
+    public void storeNonOperational(HealthAwareStore store, Exception ex) {
+        log.info("Message store became not operational.");
```

Comment:
add more information log explaing that 'TTL deletion task stopped working etc" because of store became offline.